### PR TITLE
JCLOUDS-1028: Configure idempotent methods

### PR DESCRIPTION
**Pull request by andrewgaul**
Enable POST for Atmos, S3, and Swift.

### CODE REVIEW

**Comment by andrewgaul** [comment]
@nacx Please review.

File: core/src/main/java/org/jclouds/Constants.java [hunk 1]
```diff
@@ -350,6 +350,9 @@
     */
    public static final String PROPERTY_MAX_PARALLEL_DELETES = "jclouds.max-parallel-deletes";
 
+   /** Comma-separated list of methods considered idempotent for purposes of retries. */
```
Comment (nacx) [inline core/src/main/java/org/jclouds/Constants.java]: Document the default values in the javadoc?

File: apis/atmos/src/main/java/org/jclouds/atmos/AtmosApiMetadata.java [hunk 2]
```diff
@@ -57,6 +58,7 @@ public static Properties defaultProperties() {
       Properties properties = BaseHttpApiMetadata.defaultProperties();
       properties.setProperty(PROPERTY_REGIONS, "DEFAULT");
       properties.setProperty(PROPERTY_USER_METADATA_PREFIX, "X-Object-Meta-");
+      properties.setProperty(Constants.PROPERTY_IDEMPOTENT_METHODS, "DELETE,GET,HEAD,OPTIONS,POST,PUT");
```
Comment (nacx) [inline apis/atmos/src/main/java/org/jclouds/atmos/AtmosApiMetadata.java]: [nit] Use a static import like in the other properties (consider this also int he other classes)

**Comment by nacx** [comment]
Just a couple very minor comments. Looks good to me. Thanks @andrewgaul !

File: apis/atmos/src/main/java/org/jclouds/atmos/AtmosApiMetadata.java (hunk 2 above)
Comment (andrewgaul) [inline apis/atmos/src/main/java/org/jclouds/atmos/AtmosApiMetadata.java]: Done.

File: core/src/main/java/org/jclouds/Constants.java (hunk 1 above)
Comment (andrewgaul) [inline core/src/main/java/org/jclouds/Constants.java]: Done.

**Comment by andrewgaul** [comment]
Addressed all outstanding review comments.

**Pull request merged**